backend/dataall/modules/datasets/aws/s3_dataset_client.py

```diff
@@ -32,3 +35,80 @@ def get_file_upload_presigned_url(self, data):
             return json.dumps(response)
         except ClientError as e:
             raise e
+
+
+class S3BucketPolicyClient:
```

Comment:
I'd include maybe `Dataset` in the class name to better differentiate it from `S3Client` class.

Or, as I see in this class only the `dataset.S3BucketName`, `dataset.datasetUri ` and `dataset.AwsAccountId `members are used from Dataset. If you would like to, you could generalize these 2 methods and put them in the `S3Client` class itself, then use them within Datasets providing the above mentioned members.

Reply:
I have added `Dataset` to the name.
I have split those two `S3Client` and `S3DatasetBucketPolicyClient`, because they use different boto3 clients:
S3Client uses an additional param `config=Config(signature_version='s3v4', s3={'addressing_style': 'virtual'}),`. it might have been nothing and `S3DatasetBucketPolicyClient` could use with config without issues, but I was not sure and created a new class